src/pyodide/python-entrypoint-helper.js

```diff
@@ -173,27 +173,37 @@ async function preparePython() {
   return await getMainModule();
 }
 
-
-export default {
-  async fetch(request, env, ctx) {
+function makeHandler(pyHandlerName) {
+  return async function (...args) {
     try {
       const mainModule = await preparePython();
-      if (mainModule.on_fetch === undefined) {
-        throw new Error("Python Worker should define an on_fetch method");
-      }
-      return await mainModule.on_fetch.callRelaxed(request, env, ctx);
+      return await mainModule[pyHandlerName].callRelaxed(...args);
     } catch (e) {
       console.warn(e.stack);
       throw e;
     }
-  },
-  async test(ctrl, env, ctx) {
-    try {
-      const mainModule = await preparePython();
-      return await mainModule.test.callRelaxed(ctrl, env, ctx);
-    } catch (e) {
-      console.warn(e.stack);
-      throw e;
+  };
+}
+
+const handlers = {};
+
+if (IS_WORKERD) {
+  handlers.fetch = makeHandler("on_fetch");
+  handlers.test = makeHandler("test");
+} else {
+  for (const handlerName of [
+    "fetch",
```

Comment:
Why is it fetch instead of on_fetch here?

Reply:
Oops meant to do `"on_" +` instead of `"py_" +`  a couple lines down. These are the javascript names.